extensions/spring-cloud-config-client/runtime/src/main/java/io/quarkus/spring/cloud/config/client/runtime/credentials/SpringCloudConfigClientOidcCredentialsProvider.java

```diff
@@ -0,0 +1,110 @@
+package io.quarkus.spring.cloud.config.client.runtime.credentials;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import io.quarkus.oidc.client.OidcClient;
+import io.quarkus.oidc.client.OidcClientConfig;
+import io.quarkus.oidc.client.runtime.OidcClientImpl;
+import io.quarkus.oidc.common.runtime.OidcCommonConfig;
+import io.quarkus.oidc.common.runtime.OidcCommonUtils;
+import io.quarkus.oidc.common.runtime.OidcConstants;
+import io.quarkus.spring.cloud.config.client.runtime.SpringCloudClientCredentialsProvider;
+import io.quarkus.spring.cloud.config.client.runtime.SpringCloudConfigClientConfig;
+import io.vertx.mutiny.core.MultiMap;
+import io.vertx.mutiny.ext.web.client.HttpRequest;
+import io.vertx.mutiny.ext.web.client.WebClient;
+
+public class SpringCloudConfigClientOidcCredentialsProvider implements SpringCloudClientCredentialsProvider {
+
+    private static final String OIDC_CLIENT_ID = "SpringCloudConfigOidcClient";
+
+    OidcClient oidcClient;
+
+    public SpringCloudConfigClientOidcCredentialsProvider(SpringCloudConfigClientConfig clientConfig, WebClient webClient) {
+        OidcClientConfig oidcClientConfig = clientConfig.oidc;
+
+        if (!oidcClientConfig.getId().isPresent()) {
+            oidcClientConfig.setId(OIDC_CLIENT_ID);
+        }
+
+        String authServerUriString = OidcCommonUtils.getAuthServerUrl(oidcClientConfig);
+        String tokenRequestUri;
+        if (!oidcClientConfig.discoveryEnabled) {
+            tokenRequestUri = OidcCommonUtils.getOidcEndpointUrl(authServerUriString, oidcClientConfig.tokenPath);
+        } else {
+            tokenRequestUri = OidcCommonUtils
+                    .discoverMetadata(webClient, authServerUriString, clientConfig.connectionTimeout.toMillis())
+                    .onItem().transform(json -> json.getString("token_endpoint")).await().indefinitely();
+        }
+
+        String grantType = oidcClientConfig.grant.getType().getGrantType();
```

Comment:
Hi @glefloch  Spring Cloud must be supporting a specific grant type only, probably `client_credentials` - `OidcClient` needs to support different grants - but it will fail here unless it is the grant supported by Spring Cloud, so IMHO it makes sense to throw a configuration error if it is not the one recognized by Spring Cloud